client/my-sites/upgrades/checkout/transaction-steps-mixin.jsx

```diff
@@ -87,7 +87,11 @@ var TransactionStepsMixin = {
 
 		switch ( step.name ) {
 			case 'input-validation':
-				notices.info( ( isFree( cart ) || cartItems.hasFreeTrial( cart ) ) ? this.translate( 'Submitting' ) : this.translate( 'Submitting payment' ) );
+				if ( cartItems.hasFreeTrial( cart ) ) {
+					break;
```

Comment:
Super minor, but avoiding `else` with

```js
if ( ! cartItems.hasFreeTrial( cart ) ) {
  notices.info( ... );
}
break;
```

might be easier to read here. What do you think?